ktorm-support-mysql/src/test/kotlin/me/liuwj/ktorm/support/mysql/MySqlTest.kt

```diff
@@ -232,4 +235,28 @@ class MySqlTest : BaseTest() {
 
         assert(name == "VINCE")
     }
+
+    @Test
+    fun testSumIf() {
+        val countRich = database.from(Employees)
+                .select(
+                        sum(IF((Employees.salary greaterEq 100L).cast(BooleanSqlType),
+                                then = ArgumentExpression(1, IntSqlType),
+                                otherwise = ArgumentExpression(0, IntSqlType)))
```

Comment:
We can have a overloaded version for `Int`, `fun IF(condition: ColumnDeclaring<Boolean>, then: Int, otherwise: Int)`, then we don't have to wrap the arguments as `ArgumentExpression` manually. 

Also, overloaded versions for frequent-used types are neccesary, like `Int`, `Long`, `Double`, `String`, etc.

Reply:
@vincentlauvlwj FYI, in this point, at first I created `IF` function (uppercase).
However, 'detekt' gave me error because of restriction that function name should be started with lower cases.
So, I changed `IF` to `if`, and use back-quote.